src/lib.rs

```
@@ -1807,7 +1807,22 @@ fn parse_seek_head<R: Read + Seek>(
                 if let ElementData::Location { offset, size } = entry_data {
                     let seek_fields = collect_children(&mut file, *offset, *size)?;
                     if let Ok(seek_entry) = SeekEntry::new(&mut file, &seek_fields) {
-                        seek_head.insert(seek_entry.id, segment_data_offset + seek_entry.offset);
+                        if seek_entry.id == ElementId::SeekHead {
+                            file.seek(SeekFrom::Start(segment_data_offset + seek_entry.offset))?;
+                            let (_element_id, element_data) = next_element(file)?;
+                            assert!(_element_id == ElementId::SeekHead);
```

It this error really unrecoverable? Shouldn't we return an error?

---

I added the assert during experimentation, to check than the element found is the one expected
Effectively, an error should be more appropriate